opts/options.go

```diff
@@ -83,12 +86,46 @@ var Defaults = func(o *Options) error {
 	o.RoutingTable.AutoRefresh = true
 	o.MaxRecordAge = time.Hour * 36
 
-	o.BucketSize = 20
+	o.BucketSize = defaultBucketSize
 	o.Concurrency = 3
 
 	return nil
 }
 
+// UnsetDefaults sets default DHT options. It is applied after Defaults and any options passed to the constructor in
+// order to allow for defaults that are based on other set options.
+func UnsetDefaults(o *Options) error {
```

Comment:
Let's call this `applyFallbacks` and put it in the main DHT pacakge. Users should never actually apply it.